libwacom/libwacom.c

```diff
@@ -1362,7 +1362,17 @@ libwacom_get_height(const WacomDevice *device)
 LIBWACOM_EXPORT WacomClass
 libwacom_get_class(const WacomDevice *device)
 {
-	return device->cls;
+	if (device->cls != WCLASS_UNKNOWN) {
+		return device->cls;
+	}
+	if (device->integration_flags & WACOM_DEVICE_INTEGRATED_DISPLAY &&
+		device->integration_flags & WACOM_DEVICE_INTEGRATED_SYSTEM) {
+		return WCLASS_ISDV4;
+	}
+	if (device->integration_flags & WACOM_DEVICE_INTEGRATED_DISPLAY) {
+		return WCLASS_CINTIQ;
+	}
```

Comment:
```suggestion
	unsigned int flags = device->integration_flags & (WACOM_DEVICE_INTEGRATED_DISPLAY|WACOM_DEVICE_INTEGRATED_SYSTEM);
	
	if (flags == (WACOM_DEVICE_INTEGRATED_DISPLAY|WACOM_DEVICE_INTEGRATED_SYSTEM)) {
		return WCLASS_ISDV4;
	}
	if (flags == WACOM_DEVICE_INTEGRATED_DISPLAY) {
		return WCLASS_CINTIQ;
	}
```
is i think a bit more readable. 

(oh, and I always like to have `()` around any bitwise operations, it makes the code a bit more readable)